### PR TITLE
NOTICK - Fix concurrency issue in flaky test

### PR DESCRIPTION
**Pull request by kyriathar**

### CODE REVIEW

File: client/rpc/src/integration-test/kotlin/net/corda/client/rpcreconnect/CordaRPCClientReconnectionTest.kt [hunk 1]
```diff
@@ -469,13 +469,13 @@ class CordaRPCClientReconnectionTest {
                 val clientId = UUID.randomUUID().toString()
                 val flowHandle = rpcOps.startFlowWithClientId(clientId, ::ThrowingFlow)
 
-                var erroredCounter = 0
+                val erroredCounter = AtomicInteger(0)
                 flowHandle.returnValue.doOnError {
-                    erroredCounter++
+                    erroredCounter.incrementAndGet()
```
Comment (blsemo) [inline client/rpc/src/integration-test/kotlin/net/corda/client/rpcreconnect/CordaRPCClientReconnectionTest.kt]: I would recommend using `.getAndIncrement()` - it has one less operation on a result that gets discarded anyway.

**Review by blsemo — COMMENTED** [review]
Looks good in principle, just one small remark.

File: client/rpc/src/integration-test/kotlin/net/corda/client/rpcreconnect/CordaRPCClientReconnectionTest.kt (hunk 1 above)
Comment (kyriathar) [reply client/rpc/src/integration-test/kotlin/net/corda/client/rpcreconnect/CordaRPCClientReconnectionTest.kt]: Done.

**Review by blsemo — APPROVED** [review]
LGTM

**Review by vkolomeyko — APPROVED** [review]
LGTM

**Pull request merged**